packages/teleport-plugin-common/src/builders/ast-builders.ts

```diff
@@ -182,6 +182,20 @@ export const createStateChangeStatement = (statement: UIDLStateModifierEvent, t
   )
 }
 
+export const createDynamicStyleStringsToTemplates = async (
+  str: string
+): Promise<types.JSXExpressionContainer> => {
+  const parse = await import('@babel/parser').then((mod) => mod.parse)
+  const result = parse('<style>{`' + str.trim() + '`}</style>', {
+    sourceType: 'module',
+    plugins: ['jsx'],
+  })
+  const expressionConntainer = (
+    (result.program.body[0] as types.ExpressionStatement).expression as types.JSXElement
+  ).children[0] as types.JSXExpressionContainer
+  return expressionConntainer
+}
+
```

Comment:
The upgrade in typescript has really found some deep issues that we didn't notice before. One of it is the following. We used to build strings like this
```js
.component-with-dyamic {
  width: ${props.width}
}
```
As a string and then pass the string to `generateStyledJSXTag` ad convert it to template literal. Building template literals with dynamic values inside is very tricky. So instead of re-working the whole generation process. 

I used `@babel/parser` and then used the parser to parse this string, and take the ast ad inject it back to the code generators AST. It's more like string -> AST -> string conversion.

Not the best, but creating direct AST's for template literals with strings is very very tricky. So, this is the easiest oe